api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -911,3 +907,17 @@ message UpstreamConnectionOptions {
   // If set then set SO_KEEPALIVE on the socket to enable TCP Keepalives.
   core.v3.TcpKeepalive tcp_keepalive = 1;
 }
+
+message TrackOptionalClusterStats {
+  // If timeout_budgets is true, the :ref:`timeout budget histograms
+  // <config_cluster_manager_cluster_stats_timeout_budgets>` will be published for each
+  // request. These show what percentage of a request's per try and global timeout was used. A value
+  // of 0 would indicate that none of the timeout was used or that the timeout was infinite. A value
+  // of 100 would indicate that the request took the entirety of the timeout given to it.
+  bool timeout_budgets = 1;
+
+  // If request_response_sizes is true, then the :ref:`histograms
+  // <config_cluster_manager_cluster_stats_request_response_sizes>`  tracking header and body sizes
+  // of requests and responses will be published.
+  bool request_response_sizes = 2;
```

Comment:
we may need more granular flags here, would this flag enable tracking of 4 histograms at same time? Some users may be for example only interested in tracking requests related stats.

Reply:
Currently, It enables all histograms under `struct ClusterRequestResponseSizeStats`. Yes, we might need more granularity.

Reply:
I think we should at least split it on req/resp path.

Reply:
I am Ok with separating the stats into `RequestSizeStats` and `ResponseSizeStats`. 
@jmarantz views on this?

Reply:
I'm OK with that but I don't see a strong need for that much fine-grained control. My reasoning is that by default I think Envoy should run lean, with minimal overhead/cluster. I mildly favor keeping the control coarser for simplicity, but I'm OK either way.

If you want to zoom in on some aspect on a per-cluster basis, you probably want to collect a bunch of data and you may not be sure what you want to to look at, but you are willing to spend memory to do it. You probably won't leave this enabled forever anyway, but you might as learn as much as you can.

For similar reasons I don't typically debug programs compiling only *some* of the files with "-g" (I used to do that when I was in a much more memory-constrained environment).

Reply:
Thanks for the detailed thought. 
With that in mind, now I also think it makes sense to leave the granularity as it is.